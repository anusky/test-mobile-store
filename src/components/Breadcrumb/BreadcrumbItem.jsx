import ChevronRight from "../Icons/ChevronRight";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const BreadcrumbItem = ({ crumb, isLast }) => (
  <>
    <li>
      {isLast ? (
        <span className="breadcrumb-item breadcrumb-item--last">
          {crumb.content}
        </span>
      ) : (
        <Link to={crumb?.url}>{crumb?.content}</Link>
      )}
    </li>
    {!isLast && (
      <li>
        <ChevronRight />
      </li>
    )}
  </>
);

export default BreadcrumbItem;
BreadcrumbItem.propTypes = {
  crumb: PropTypes.shape({
    url: PropTypes.string,
    content: PropTypes.node,
  }),
  isLast: PropTypes.bool.isRequired,
};
