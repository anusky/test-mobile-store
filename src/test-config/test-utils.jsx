import { render } from "@testing-library/react";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import PropTypes from "prop-types";

import productInfo from "./__fixtures__/product.json";
import productList from "./__fixtures__/productList.json";

const customRender = (ui, options = {}) => {
  const { routerInitialIndex = 0 } = options;

  const AllTheProviders = ({ children }) => {
    const routes = [
      {
        path: "/",
        element: children,
        loader: () => productList,
        handle: {
          crumb: () => ({ url: "/", content: "home route" }),
        },
        children: [
          {
            path: "/:productId",
            element: children,
            id: "product-page",
            loader: () => productInfo,
            handle: {
              crumb: () => ({ url: "/", content: "product route" }),
            },
          },
        ],
      },
    ];

    const router = createMemoryRouter(routes, {
      initialEntries: ["/", "/123"],
      initialIndex: routerInitialIndex,
    });
    return <RouterProvider router={router}>{children}</RouterProvider>;
  };

  AllTheProviders.propTypes = {
    children: PropTypes.node.isRequired,
  };

  return render(ui, { wrapper: AllTheProviders, ...options });
};

export * from "@testing-library/react";
export { customRender as render };
