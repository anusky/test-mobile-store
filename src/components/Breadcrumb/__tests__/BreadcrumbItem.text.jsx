import { act } from "react-dom/test-utils";
import { render, screen } from "src/test-config/test-utils";

import BreadcrumbItem from "../BreadcrumbItem";

describe("BreadcrumbItem coverage", () => {
  it("renders correctly", async () => {
    render(
      <BreadcrumbItem
        crumb={{
          content: "test text",
          url: "/",
        }}
        isLast={false}
      />
    );
    expect(await screen.findByText("test text")).toBeInTheDocument();
  });

  it("Should render two listitem when it is not the last item", async () => {
    render(
      <BreadcrumbItem
        crumb={{
          content: "test text",
          url: "/",
        }}
        isLast={false}
      />
    );

    const totalListItems = await screen.findAllByRole("listitem");
    expect(totalListItems.length).toBe(2);
  });
  it("Should render a single listitem when it is the last item", async () => {
    render(
      <BreadcrumbItem
        crumb={{
          content: "test text",
        }}
        isLast={true}
      />
    );

    const totalListItems = await screen.findAllByRole("listitem");
    expect(totalListItems.length).toBe(1);
  });

  it("snapshot for BreadcrumbItem isLast is true", async () => {
    let container;
    await act(() => {
      container = render(
        <BreadcrumbItem
          crumb={{
            content: "test text",
          }}
          isLast={true}
        />
      );
    });
    expect(container.asFragment()).toMatchSnapshot();
  });
  it("snapshot for BreadcrumbItem when isLast is false", async () => {
    let container;
    await act(() => {
      container = render(
        <BreadcrumbItem
          crumb={{
            content: "test text",
            url: "/",
          }}
          isLast={false}
        />
      );
    });
    expect(container.asFragment()).toMatchSnapshot();
  });
});
