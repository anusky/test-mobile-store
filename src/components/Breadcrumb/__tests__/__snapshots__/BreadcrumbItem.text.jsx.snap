// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbItem coverage snapshot for BreadcrumbItem isLast is true 1`] = `
<DocumentFragment>
  <li>
    <span
      class="breadcrumb-item breadcrumb-item--last"
    >
      test text
    </span>
  </li>
</DocumentFragment>
`;

exports[`BreadcrumbItem coverage snapshot for BreadcrumbItem when isLast is false 1`] = `
<DocumentFragment>
  <li>
    <a
      href="/"
    >
      test text
    </a>
  </li>
  <li>
    <svg
      class="w-5 h-5"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M16.28 11.47a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 01-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 011.06-1.06l7.5 7.5z"
        fill-rule="evenodd"
      />
    </svg>
  </li>
</DocumentFragment>
`;
